From 50abf4695c2faa0537ad53c8c6d85210ec639fa0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 7 Jul 2017 03:09:38 +0100
Subject: [PATCH 18/33] Quick Adapta prettiness hacks

 - Force dark theme
 - Drop the derpy toolbar
 - Stick icon + label in titlebar itself
 - Drop derpy welcome page

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py     | 29 ++++++++++++++++++++++++-----
 os_installer2/pages/basepage.py | 19 -------------------
 2 files changed, 24 insertions(+), 24 deletions(-)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index ade27bd..090c13b 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -103,6 +103,9 @@ class MainWindow(Gtk.ApplicationWindow):
     can_quit = True
     is_final_step = False
 
+    image_step = None
+    label_step = None
+
     def quit_handler(self, w, udata=None):
         """ Ensure quit stuff is sane ... """
         if not self.can_quit:
@@ -113,11 +116,23 @@ class MainWindow(Gtk.ApplicationWindow):
         Gtk.ApplicationWindow.__init__(self, application=app)
         self.application = app
 
-        header = Gtk.HeaderBar.new()
-        header.set_show_close_button(True)
-        self.set_titlebar(header)
+        Gtk.Settings.get_default().set_property("gtk-application-prefer-dark-theme", True)
+
+        self.headerbar = Gtk.HeaderBar.new()
+        self.headerbar.set_show_close_button(True)
+        self.set_titlebar(self.headerbar)
+
+        self.image_step = Gtk.Image.new_from_icon_name("system-software-install", Gtk.IconSize.LARGE_TOOLBAR)
+        self.image_step.set_pixel_size(32)
+        self.image_step.set_property("margin", 4)
+        self.label_step = Gtk.Label.new("")
+
+        box = Gtk.Box.new(Gtk.Orientation.HORIZONTAL, 0)
+        box.pack_start(self.label_step, False, False, 0)
+        self.headerbar.set_custom_title(box)
+        self.headerbar.pack_start(self.image_step)
 
-        self.set_title("Installer")
+        self.set_title("")
         self.set_icon_name("system-software-install")
         self.connect("delete-event", self.quit_handler)
 
@@ -135,7 +150,6 @@ class MainWindow(Gtk.ApplicationWindow):
         self.installer_stack = Gtk.Stack()
         self.installer_page.pack_start(self.installer_stack, True, True, 0)
 
-        self.stack.add_named(InstallerWelcomePage(self), "welcome")
         self.installer_stack.set_transition_type(ltr)
         self.stack.add_named(self.installer_page, "install")
 
@@ -273,6 +287,11 @@ class MainWindow(Gtk.ApplicationWindow):
         else:
             self.set_can_previous(True)
         page.prepare(self.info)
+        mk = u"<span font-size='x-large'>{}</span>".format(page.get_title())
+        self.label_step.set_markup(mk)
+        self.image_step.set_from_icon_name(page.get_icon_name(),
+                                           Gtk.IconSize.LARGE_TOOLBAR)
+        self.image_step.set_pixel_size(32)
         self.installer_stack.set_visible_child_name(page.get_name())
 
     def set_can_previous(self, can_prev):
diff --git a/os_installer2/pages/basepage.py b/os_installer2/pages/basepage.py
index 4c6cc12..eb7f984 100644
--- a/os_installer2/pages/basepage.py
+++ b/os_installer2/pages/basepage.py
@@ -20,25 +20,6 @@ class BasePage(Gtk.VBox):
     def __init__(self):
         Gtk.VBox.__init__(self)
 
-        self.title = Gtk.Label(u"<span font-size='xx-large'>{}</span>".format(
-                               self.get_title()))
-        self.title.set_use_markup(True)
-
-        self.image = Gtk.Image()
-        self.image.set_from_icon_name(self.get_icon_name(),
-                                      Gtk.IconSize.DIALOG)
-        self.image.set_padding(10, 10)
-
-        header = Gtk.HBox()
-        header.pack_start(self.image, False, False, 0)
-        header.pack_start(self.title, False, False, 0)
-
-        header.get_style_context().add_class("primary-toolbar")
-
-        self.pack_start(header, False, True, 0)
-        self.image.set_margin_start(20)
-        self.title.set_margin_start(10)
-
     def get_title(self):
         return None
 
-- 
2.13.3

